ui/src/components/Banner.jsx

```diff
@@ -0,0 +1,272 @@
+
+import logo from './logo.png';
+import './Banner.scss';
+
+export default function Banner() {
+  useEffect(() => {
+    const image = main()
+    return () => { image.remove() }
+  }, [])
+
+  return <canvas className="Banner" id="glcanvas"></canvas>
+}
+
+function main() {
+  const image = new Image();
+  image.src = logo;
+  image.onload = function() { render(image); }
+  return image
+}
+
+function render(image) {
+  const canvas = document.getElementById('glcanvas');
+  canvas.setAttribute('width', image.width);
+  canvas.setAttribute('height', image.height);
+
+  if (window.location.href.startsWith('file:///')) {
+    alert(`You can't see the cool logo if you open this file directly from the filesystem :-(\n\nCheck docs/design/README.md for more information.`);
+    return;
+  }
+
+  const gl = canvas.getContext('webgl');
+
+  const vertexShader = createShader(gl, gl.VERTEX_SHADER, vertexShaderSource);
+  const fragmentShader = createShader(gl, gl.FRAGMENT_SHADER, fragmentShaderSource);
+  const program = createProgram(gl, vertexShader, fragmentShader);
+
+  if (!program) {
+    throw new Error('error compiling program');
+  }
+
+  const positionAttributeLocation = gl.getAttribLocation(program, 'a_position');
+  const canvasResolutionUniformLocation = gl.getUniformLocation(program, 'u_vertexResolution');
+  const imageResolutionUniformLocation = gl.getUniformLocation(program, 'u_imageResolution');
+  const timeUniformLocation = gl.getUniformLocation(program, 'iTime');
+
+  // Rectangle with the same size as the image in positionBuffer
+  const positionBuffer = gl.createBuffer();
+  gl.bindBuffer(gl.ARRAY_BUFFER, positionBuffer);
+  setRectangle(gl, 0, 0, image.width, image.height);
+
+  // Upload image into texture
+  const texture = gl.createTexture();
+  gl.bindTexture(gl.TEXTURE_2D, texture);
+  gl.texParameteri(gl.TEXTURE_2D, gl.TEXTURE_WRAP_S, gl.CLAMP_TO_EDGE);
+  gl.texParameteri(gl.TEXTURE_2D, gl.TEXTURE_WRAP_T, gl.CLAMP_TO_EDGE);
+  gl.texParameteri(gl.TEXTURE_2D, gl.TEXTURE_MIN_FILTER, gl.NEAREST);
+  gl.texParameteri(gl.TEXTURE_2D, gl.TEXTURE_MAG_FILTER, gl.NEAREST);
+  gl.texImage2D(gl.TEXTURE_2D, 0, gl.RGBA, gl.RGBA, gl.UNSIGNED_BYTE, image);
+
+  requestAnimationFrame(drawScene);
+
+  function drawScene(now) {
+    now *= 0.001;
+
+    gl.viewport(0, 0, image.width, image.height);
+    gl.clearColor(0.149, 0.145, 0.141, 1);
+    gl.clear(gl.COLOR_BUFFER_BIT);
+
+    gl.useProgram(program);
+
+    // Bind attributes to buffers
+    gl.enableVertexAttribArray(positionAttributeLocation);
+    gl.bindBuffer(gl.ARRAY_BUFFER, positionBuffer);
+    gl.vertexAttribPointer(positionAttributeLocation, 2, gl.FLOAT, false, 0, 0);
+
+    gl.uniform2f(canvasResolutionUniformLocation, image.width, image.height);
+    gl.uniform2f(imageResolutionUniformLocation, image.width, image.height);
+    gl.uniform1f(timeUniformLocation, now);
+    gl.drawArrays(gl.TRIANGLES, 0, 6);
+
+    requestAnimationFrame(drawScene);
```

Comment:
this needs to be stopped when the component is unmounted, see https://css-tricks.com/using-requestanimationframe-with-react-hooks/